sdk/src/main/java/io/opentelemetry/sdk/trace/Sampler.java

```diff
@@ -68,27 +68,36 @@ Decision shouldSample(
    */
   String getDescription();
 
+  /** A decision on whether a span should be recorded, recorded and sampled or not recorded. */
+  enum Decision {
+    NOT_RECORD,
+    RECORD,
+    RECORD_AND_SAMPLED,
+  }
+
   /**
-   * Sampling decision returned by {@link Sampler#shouldSample(SpanContext, TraceId, String, Kind,
+   * Sampling result returned by {@link Sampler#shouldSample(SpanContext, TraceId, String, Kind,
    * ReadableAttributes, List)}.
    *
    * @since 0.1.0
    */
-  interface Decision {
+  interface SamplingResult {
 
     /**
-     * Return sampling decision whether span should be sampled or not.
+     * Return decision on whether a span should be recorded, recorded and sampled or not recorded.
      *
-     * @return sampling decision.
-     * @since 0.1.0
+     * @return sampling result.
+     * @since 0.7.0
      */
+    Decision getDecision();
+
     boolean isSampled();
```

Comment:
Can we remove this?

Reply:
It's no longer required in the spec so it would seem like it's no longer required

https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#sampler